src/libraries/System.Private.CoreLib/src/System/Globalization/CalendarData.iOS.cs

```diff
@@ -0,0 +1,111 @@
+            this.saMonthGenitiveNames = NormalizeMonthArray(GetCalendarInfoNative(localeName, calendarId, CalendarDataType.MonthGenitiveNames).Split("||"), calendarId, ref leapHebrewMonthName);
+            this.saAbbrevMonthGenitiveNames = NormalizeMonthArray(GetCalendarInfoNative(localeName, calendarId, CalendarDataType.AbbrevMonthGenitiveNames).Split("||"), calendarId, ref leapHebrewMonthName);
+
+            this.saEraNames = NormalizeEraNames(calendarId, GetCalendarInfoNative(localeName, calendarId, CalendarDataType.EraNames).Split("||"));
+            this.saAbbrevEraNames = Array.Empty<string>();//NormalizeEraNames(calendarId, GetCalendarInfoNative(localeName, calendarId, CalendarDataType.AbbrevEraNames).Split("||"));
+
+            return this.sNativeName != null && this.saShortDates != null && this.saLongDates != null && this.saYearMonths != null &&
+                   this.saDayNames != null && this.saAbbrevDayNames != null && this.saSuperShortDayNames != null && this.saMonthNames != null &&
+                   this.saAbbrevMonthNames != null && this.saMonthGenitiveNames != null && this.saAbbrevMonthGenitiveNames != null &&
+                   this.saEraNames != null && this.saAbbrevEraNames != null;
+        }
+
+        private static string[] NormalizeEraNames(CalendarId calendarId, string[]? eraNames)
+        {
+            // .NET expects that only the Japanese calendars have more than 1 era.
+            // So for other calendars, only return the latest era.
+            if (calendarId != CalendarId.JAPAN && calendarId != CalendarId.JAPANESELUNISOLAR && eraNames?.Length > 0)
+            {
+                string[] latestEraName = new string[] { eraNames![eraNames.Length - 1] };
+                eraNames = latestEraName;
+            }
+
+            return eraNames ?? Array.Empty<string>();
+        }
+
+        private static string[] NormalizeMonthArray(string[] months, CalendarId calendarId, ref string? leapHebrewMonthName)
+        {
+            if (months.Length == 13)
+                return months;
+
+            string[] normalizedMonths = new string[13];
+            // the month-name arrays are expected to have 13 elements.  If only returns 12, add an
+            // extra empty string to fill the array.
+            if (months.Length == 12)
+            {
+                normalizedMonths[12] = "";
+                months.CopyTo(normalizedMonths, 0);
+                return normalizedMonths;
+            }
+
+            if (months.Length > 13)
+            {
+                Debug.Assert(calendarId == CalendarId.HEBREW && months.Length == 14);
+
+                if (calendarId == CalendarId.HEBREW)
+                {
+                    leapHebrewMonthName = months[13];
+                }
+                for (int i = 0; i < 13; i++)
+                {
+                    normalizedMonths[i] = months[i];
+                }
+                return normalizedMonths;
+            }
```

Comment:
I think we can just return `months` instead of allocating and copying to a new array.

```
Debug.Assert(months.Length <= 14);
if (months.Length == 14)
{
    Debug.Assert(calendarId == CalendarId.HEBREW);
    leapHebrewMonthName = months[13];
}

if (months.Length >= 13)
    return months;
}
```
And collapse the top early return into this logic too

Reply:
We can't return months in case `months.Length == 14` , we need to return array with 13 values.